src/sysfiles.h

```diff
@@ -139,23 +139,36 @@ extern const Char * SyWinCmd (
 **  routines  from   allocating their  buffers  using  'malloc',  which would
 **  otherwise confuse Gasman.
 */
+
+// The type of file stored in a 'SYS_SY_BUF'
+typedef enum {
+    unused_socket,  // Socket is free
+    raw_socket,     // Plain UNIX socket stored in 'fp'
+    pipe_socket,    // UNIX socket attached to a pipe stored in 'fp'
+    gzip_socket     // A gzFile handled by zlib stored in 'gzfp'
+} GAPSocketType;
+
 typedef struct {
-  int         fp;          /* file descriptor for this file. This is used
-                              for input */
-  int         echo;        /* file descriptor for the echo. This is used
-                              for output. Is set to 0 if read-only file */
-  UInt        pipe;        /* file is really a pipe */
-  FILE       *pipehandle;  /* for pipes we need to remember the file handle */
-  UInt        ateof;       /* set to 1 by any read operation that hits eof
+    /* gzfp is used if type == gzip_socket
+    ** Stored as a void* to avoid including zlib.h */
+    void* gzfp;
```

Comment:
Formatting: space before `*` (but we can also fix this afterwards -- indeed, with your PR merged, I think we are close to making `syBuf` private to `sysfiles.h`, then we could move this out of the header and into the .c file, and take care of formatting in this struct overall)

Reply:
Yes, we could try moving this into sysfiles.h later. I did at some point run clang-format on this part, but it totally changed the whole struct and I wanted it to be clear what I'd added/changed.